### PR TITLE
Add XUnit report support

### PR DESCRIPTION
**Pull request by perrich**
Add management for XUnit XML file. Should work with 1.x and 2.x versions.
XUnit is a great alternative to NUnit. See http://xunit.github.io/

### CODE REVIEW

File: src/main/java/org/sonar/plugins/dotnet/tests/XUnitTestResultsFileParser.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          xmlParserHelper.close();
+        }
+      }
+    }
+
+    private void checkRootTag() {
+      String rootTag = xmlParserHelper.nextTag();
+    
+      if ("assemblies".equals(rootTag)) {
+        isSecondVersion = true;
```
Comment (dbolkensteyn) [inline src/main/java/org/sonar/plugins/dotnet/tests/XUnitTestResultsFileParser.java]: I'm seeing a `<assemblies>` tag as root even when `-xmlv1` is used

File: src/main/java/org/sonar/plugins/dotnet/tests/XUnitTestResultsFileParser.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        rootTag = xmlParserHelper.nextTag();
+      }
+
+      if (!"assembly".equals(rootTag)) {
```
Comment (dbolkensteyn) [inline src/main/java/org/sonar/plugins/dotnet/tests/XUnitTestResultsFileParser.java]: there can be multiple `<assembly>` tags within the `<assemblies>` one. The metrics need to be summed to get the overall numbers. This happens when you pass several test assemblies to the xunit runner.

If you chose the `-nunit` format, then you'll see all metrics already summed in the `test-results` root tag

File: src/test/resources/xunit/invalid_total.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<!--This file represents the results of running a test suite-->
+<assembly name="c:\git\sonar-xunit\Sonar.XUnit\bin\Debug\Sonar.XUnit.dll" run-date="2014-08-15" run-time="09:02:13" time="0.603" total="invalid" passed="0" skipped="1" failed="0" environment="32-bit .NET 4.0.30319.34209" test-framework="xUnit.net 1.9.2.1705">
```
Comment (dbolkensteyn) [inline src/test/resources/xunit/invalid_total.xml]: I'll also give xUnit 1.x a try

**Comment by dbolkensteyn** [comment]
This is going to be merged as part of https://github.com/SonarSource/sonar-dotnet-tests-library/pull/15

**Comment by perrich** [comment]
Thank you for this great work!

**Pull request closed without merging**